homeassistant/helpers/location.py

```diff
@@ -41,3 +48,61 @@ def closest(
             longitude,
         ),
     )
+
+
+def find_coordinates(
+    hass: HomeAssistantType, entity_id: str, recursion_history: Optional[list] = None
+) -> Optional[str]:
+    """Find the gps coordinates of the entity in the form of '90.000,180.000'."""
+    entity_state = hass.states.get(entity_id)
+
+    if entity_state is None:
+        _LOGGER.error("Unable to find entity %s", entity_id)
+        return None
+
+    # Check if the entity has location attributes
+    if has_location(entity_state):
+        return _get_location_from_attributes(entity_state)
+
+    # Check if device is in a zone
+    zone_entity = hass.states.get(f"zone.{entity_state.state}")
+    if has_location(zone_entity):  # type: ignore
+        _LOGGER.debug(
+            "%s is in %s, getting zone location", entity_id, zone_entity.entity_id  # type: ignore
+        )
+        return _get_location_from_attributes(zone_entity)  # type: ignore
+
+    # Resolve nested entity
+    if recursion_history is None:
+        recursion_history = []
+    recursion_history.append(entity_id)
+    if entity_state.state in recursion_history:
+        _LOGGER.error(
+            "Circular reference detected while trying to find coordinates of an entity. The state of %s has already been checked.",
```

Comment:
We don't end logging messages with period.

Reply:
Done